### PR TITLE
GET Environment variable service

### PR DESCRIPTION
**Pull request by vincentmigot**
And change EnvironmentMeasure date type from String to Date

### CODE REVIEW

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java [hunk 1]
```diff
@@ -42,15 +46,115 @@
...
+
+        // Get the filter query
+        BasicDBObject query = prepareSearchQuery();
+        
+        // Return the document count
+        return (int)environmentMeasureVariableCollection.count(query);
+    }
 
     @Override
     protected BasicDBObject prepareSearchQuery() {
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java]: add the class documentation with an example of query :)

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java [hunk 2]
```diff
@@ -42,15 +46,115 @@
...
+        } catch (ParseException ex) {
+            LOGGER.error("Invalid date format", ex);
+        }
+        
+        // Add filter if a sensor uri is defined
+        if (sensorUri != null) {
+            query.append(DB_FIELD_SENSOR, sensorUri);
+        }
+        
+        LOGGER.trace(getTraceabilityLogs() + " query : " + query.toString());
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java]: Change the logger.trace to logger.debug

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java [hunk 3]
```diff
@@ -42,15 +46,115 @@
...
+        
+        return query;
     }
 
     @Override
     public ArrayList<EnvironmentMeasure> allPaginate() {
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/EnvironmentDAOMongo.java]: Add method doc with the list of params used to search  the environment data. 

**To discuss with Anne** 
Maybe we could start to change the dao and directly send the search params or model to the method ? (I think that we started this new way to code in the brapi trait refactoring)

File: phis2-ws/src/main/java/phis2ws/service/resources/EnvironmentResourceService.java [hunk 4]
```diff
@@ -126,4 +138,130 @@ public Response postEnvironmentMeasures(
...
+     * Service to get environment measures
+     * @param pageSize
+     * @param page
+     * @param variable
+     * @param startDate
+     * @param endDate
+     * @param sensor
+     * @param dateSortAsc
+     * @return list of the environment measures corresponding to the search params given
+     * e.g
```
Comment (Gouga34) [inline phis2-ws/src/main/java/phis2ws/service/resources/EnvironmentResourceService.java]: @example ;)

Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/resources/EnvironmentResourceService.java]: done

**Pull request closed without merging**